src/servers/launcher/launcher-review-request.md

```diff
@@ -0,0 +1,82 @@
+
+* There's a lambda being used to `cout` a `variant` of exceptions in `main()` which may be able to be replaced with `cout`ing the `variant` directly now I implemented `operator<<` for `variant`s.
+* There's a bug when it comes to not clearing the old packet buffer, then receiving an incomplete message.  I plan to fix this by implementing `struct Packet` that will provide `std::expected<ValidatedPacket, std::string> getValidatedPacket()`, where the `ValidatedPacket` is some view onto the underlying buffer with the transport level anti corruption checks stripped out.
+* I will be splitting this into different files.  Just wanted to get it running and learn stuff along the way, first.
+* The heartbeat will record compute/memory pressure stats in the database alongside the `launcher`'s IP in the future, such that `cluster` can pick an appropriate server on which to launch new instances.  Right now there's not a great deal of reason to update the heartbeat in the database each second as I do.
+* I've shyed away from external libraries (apart from `libpq`) mostly to learn how things work "under the hood" myself, but also in a small effort to avoid "bloat".  I'm aware Boost provides things like a UUID class and networking stuff, but I'm under the impression Boost is mega-huge, and I'd like to not use stuff like that, at least for now, unless there's an insanely compelling reason to.  (For the client, I'm sure I'll use something like raylib, so external dependencies will come).
+* String interpolation for SQL instead of paramater passing - I know.  Trust me I do.  I've gone with this way _for now_ because I want to visit [sqlc](https://sqlc.dev) for repository construction, which is going to require some work, at which point I'll deal with parameter passing, prepared statements, and probably more.
+
+# Specific things that could be better that I'd like to know about.
+
+* I haven't used RAII for saving and restoring `os.flags()` in `operator<<(std::ostream&, const UUID&)`.  It seems like it might be overkill, but it also seems like exactly the sort of place I should be using RAII.  (I'd use a context manager in Python).
+* `UUID` instantiation feels weird.  Having to directly name every offset from `[0]` to `[15]`, and providing a way to unsafely create a `UUID` from `const char*`...  I don't know, I just don't like it.
+* In a couple places, I'm turning off `-Wmissing-field-initializers` when constructing `sockaddr_in`, which doesn't seem right.  I'm pretty sure I'm setting every field I need to in the structs, but the compiler still whinges.
+* I would love to know how to write `template <typenameT, typename... Ts> operator<<(std::ostream& const std::variant<T, Ts...>&)` with `auto` and concepts instead of a template.
+* I don't like being able to use `static_cast<ScopedEnum>(...)` to make nonsensical values, and thus being forced to handle that in functions accepting a scoped enum.
+* Can I usefully use constexpr in more places?
```

Comment:
You are writing a game server, whose primary task is to connect to a database and remote clients.

In other words, probably not.